docs/book/src/reference/webhook-for-core-types.md

```diff
@@ -7,47 +7,58 @@ There is an [example](https://github.com/kubernetes-sigs/controller-runtime/tree
 in controller-runtime.
 
 It is suggested to use kubebuilder to initialize a project, and then you can
-follow the steps below to add admission webhooks for core types.
+follow the steps below to add admission webhooks for core types. The example shows
+how to set up a mutating webhook following the controller-runtime's webhook builder.
 
-## Implement Your Handler
+## Implement Your Webhook
 
-You need to have your handler implements the
-[admission.Handler](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#Handler)
+You need to have your webhook to implement the
+[admission.CustomDefaulter](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#CustomDefaulter)
 interface.
 
 ```go
-type podAnnotator struct {
-	Client  client.Client
-	decoder *admission.Decoder
+type podAnnotator struct {}
+
+func (a *podAnnotator) setupWebhookWithManager(mgr ctrl.Manager) error {
+       return ctrl.NewWebhookManagedBy(mgr).
+               For(&corev1.Pod{}).
+               WithDefaulter(a).
+               Complete()
 }
 
-func (a *podAnnotator) Handle(ctx context.Context, req admission.Request) admission.Response {
-	pod := &corev1.Pod{}
-	err := a.decoder.Decode(req, pod)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
+func (a *podAnnotator) Default(ctx context.Context, obj runtime.Object) error {
+    log := logf.FromContext(ctx)
+    pod, ok := obj.(*corev1.Pod)
+    if !ok {
+        return fmt.Errorf("expected a Pod but got a %T", obj)
+    }
 
 	// mutate the fields in pod
+    if pod.Annotations == nil {
+        pod.Annotations = map[string]string{}
+    }
+    pod.Annotations["example-mutating-admission-webhook"] = "foo"
+
+    log.Info("Annotated pod")
 
-	marshaledPod, err := json.Marshal(pod)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+	return nil
```

Comment:
before it could return an error because it was patching the value 
so, if an error were faced it will be returned now I am not sure if we we are patching the value.
Did you tested it out?

Reply:
> Did you tested it out?

The code is same as in https://github.com/kubernetes-sigs/controller-runtime/blob/76d3d0826fa9dca267c70c68c706f6de40084043/examples/builtins/mutatingwebhook.go so I'm assuming it's tested by controller-runtime. FWIW, my own webhook follows the same flow also.

Reply:
It is not the same
Before it was use the client to perform the changes on the cluster side and return its result (nil OR error)
Now, it adds the value to the pod, do nothing and return nil always

Reply:
> It is not the same

I'm pretty sure it's the same. I just copied [the `Default()` func](https://github.com/kubernetes-sigs/controller-runtime/blob/76d3d0826fa9dca267c70c68c706f6de40084043/examples/builtins/mutatingwebhook.go) and compared with what is in this file and there are no functional changes.

Reply:
But the example that we have here does

```
marshaledPod, err := json.Marshal(pod)
	if err != nil {
		return admission.Errored(http.StatusInternalServerError, err)
	}
	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
```

It would not add too much value just have the sample controller-runtime right?

Reply:
Hi @mythi 

Sure, and we appreciate it a lot.
What I am trying to say is that we should update the doc to keep the same example that we have before but now addressing the controller-runtime changes. That would provide more value than change the current example to copy and paste an example from there.

Reply:
> That would provide more value than change the current example to copy and paste an example from there.

What I tried to say was that the current example is also a copy and paste and that the diff here is how controller-runtime addressed their changes in the example. I kinda understand what you're asking me to change but the counterargument is why would we want to diverge from the example given elsewhere (especially since we used to be aligned).